utils/chunker/stream.go

```diff
@@ -49,10 +54,12 @@ func (s *Stream) Read(p []byte) (n int, err error) {
 
 // Close closes reader.
 func (s *Stream) Close() error {
+	s.m.Lock()
+	defer s.m.Unlock()
 	if s.done == nil {
 		return nil
 	}
-	close(s.done)
```

Comment:
In that case, done channel is not closed anywhere and it leads to leaks.

Reply:
it is garbage collected. non-closed chans does not cause memory leaks

Reply:
True, ok for me then